### PR TITLE
Enhancement: Allow whitelisting of classes

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [x] allows whitelisting of classes

Follows https://twitter.com/marcospassos/status/1062887609062567936.

/cc @marcospassos

### CODE REVIEW

File: src/Classes/AbstractOrFinalRule.php [hunk 1]
```diff
@@ -33,6 +48,7 @@ public function getNodeType(): string
     public function processNode(Node $node, Scope $scope): array
     {
         if (!isset($node->namespacedName)
```
Comment (marcospassos) [inline src/Classes/AbstractOrFinalRule.php]: I usually prefer a safer check:
```php
        if (!$node instance of  Node\Stmt\Class_
            || \in_array($node->namespacedName->toString(), $this->excludedClassNames, true)
            || $node->isAbstract()
            || $node->isFinal()
        ) {
```

File: src/Classes/AbstractOrFinalRule.php [hunk 2]
```diff
@@ -19,6 +19,21 @@
...
+     * @var string[]
+     */
+    private $excludedClassNames;
+
+    /**
+     * @param string[] $excludedClassNames
+     */
+    public function __construct(array $excludedClassNames = [])
+    {
+        $this->excludedClassNames = \array_map(function (string $excludedClassName): string {
```
Comment (marcospassos) [inline src/Classes/AbstractOrFinalRule.php]: You can declare this closure static:
```php
static function (string $excludedClassName): string {
```

File: src/Classes/FinalRule.php [hunk 3]
```diff
@@ -19,6 +19,21 @@
...
+     * @var string[]
+     */
+    private $excludedClassNames;
+
+    /**
+     * @param string[] $excludedClassNames
+     */
+    public function __construct(array $excludedClassNames = [])
+    {
+        $this->excludedClassNames = \array_map(function (string $excludedClassName): string {
```
Comment (marcospassos) [inline src/Classes/FinalRule.php]: You can declare this closure static:
```php
static function (string $excludedClassName): string {
```

File: src/Classes/FinalRule.php [hunk 4]
```diff
@@ -32,7 +47,10 @@ public function getNodeType(): string
      */
     public function processNode(Node $node, Scope $scope): array
     {
-        if (!isset($node->namespacedName) || $node->isFinal()) {
+        if (!isset($node->namespacedName)
```
Comment (marcospassos) [inline src/Classes/FinalRule.php]: I usually prefer a safer check:
```php
        if (!$node instance of  Node\Stmt\Class_
            || \in_array($node->namespacedName->toString(), $this->excludedClassNames, true)
            || $node->isFinal()
        ) {
```

File: src/Classes/AbstractOrFinalRule.php (hunk 2 above)
Comment (localheinz) [reply src/Classes/AbstractOrFinalRule.php]: Fixing with https://github.com/localheinz/php-cs-fixer-config/pull/157.

File: src/Classes/FinalRule.php (hunk 3 above)
Comment (localheinz) [reply src/Classes/FinalRule.php]: Fixing with https://github.com/localheinz/php-cs-fixer-config/pull/157.

File: src/Classes/AbstractOrFinalRule.php (hunk 1 above)
Comment (localheinz) [reply src/Classes/AbstractOrFinalRule.php]: Sounds good, added - and throwing a `PHPStand\ShouldNotHappenException`.

**Comment by localheinz** [comment]
@localheinz 

Now that both `FinalRule` and `AbstractOrFinalRule` are configurable, this makes sense.

I'll follow up!

**Comment by localheinz** [comment]
@marcospassos 

Not entirely sure yet, I'm with Martin Fowler on that: https://martinfowler.com/bliki/FlagArgument.html.

**Comment by marcospassos** [comment]
I'm with him on that too, but I don't think those arguments are applicable here for the following reasons:
- I don't think that duplicating a whole class and test suite to avoid a boolean property worth the effort
- I believe you will have no other flags for this class, otherwise, you can end up violating the single responsibility principle. 
- Currently, it requires reading the documentation to understand the difference between them, as both serve the same propose.

**Pull request merged**